api/api.go

```diff
@@ -45,23 +64,25 @@ func CreateAndInitialiseRecipeAPI(ctx context.Context, cfg config.Configuration,
 }
 
 //NewRecipeAPI create a new Recipe API instance and register the API routes based on the application configuration.
-func NewRecipeAPI(ctx context.Context, cfg config.Configuration, router *mux.Router, dataStore store.DataStore) *RecipeAPI {
+func NewRecipeAPI(ctx context.Context, cfg config.Configuration, router *mux.Router, dataStore store.DataStore, recipePermissions AuthHandler, permissions AuthHandler) *RecipeAPI {
 	api := &RecipeAPI{
-		dataStore:        dataStore,
-		Router:           router,
-		EnableAuthImport: cfg.MongoConfig.EnableAuthImport,
+		dataStore:         dataStore,
+		Router:            router,
+		recipePermissions: recipePermissions,
+		permissions:       permissions,
+		EnableAuthImport:  cfg.MongoConfig.EnableAuthImport,
 	}
 
 	api.get("/health", api.HealthCheck)
 	api.get("/recipes", api.RecipeListHandler)
 	api.get("/recipes/{id}", api.RecipeHandler)
```

Comment:
`curl -L -X GET 'http://localhost:22300/recipes'` returns 401 Unauthorized code, my understanding is that the GET methods shouldn't need auth headers? Maybe someone else could clarify this.

Reply:
Problem resolved in commit "remove identity middleware to avoid authenticating whole API"